code/retrieval/retriever.py

```diff
@@ -185,82 +190,348 @@ def __init__(self, endpoint_name: Optional[str] = None, query_params: Optional[D
         Initialize the database client.
         
         Args:
-            endpoint_name: Optional name of the endpoint to use
+            endpoint_name: Optional name of the endpoint to use (for backward compatibility)
             query_params: Optional query parameters for overriding endpoint
         """
         self.query_params = query_params or {}
+        self.endpoint_name = endpoint_name  # Store the endpoint name
```

Comment:
It looks like you still expose the query_params in the init method but you don't use them anywhere or override the defaults with them anymore, which is a functionality change.  Can you please fix this?  Either use them or get rid of them.